src/hooks/useActiveChainId.ts

```diff
@@ -0,0 +1,9 @@
+import { useNetwork } from "wagmi"
+
+import { useActiveChain } from "@/store/activeChain"
+
+export default function useActiveChainId() {
```

Comment:
This hook will always give you the chainId that you should use. 

Prioritizing connected wallet before the disconnected chainId.